clang/lib/Sema/SemaExpr.cpp

```diff
@@ -5813,6 +5813,24 @@ static TypoCorrection TryTypoCorrectionForCall(Sema &S, Expr *Fn,
   return TypoCorrection();
 }
 
+static bool isParenthetizedAndQualifiedAddressOfExpr(Expr *Fn) {
+  if (!isa<ParenExpr>(Fn))
+    return false;
+
+  Fn = Fn->IgnoreParens();
+  auto *UO = dyn_cast<UnaryOperator>(Fn);
+  if (!UO)
+    return false;
+  assert(cast<UnaryOperator>(Fn)->getOpcode() == UO_AddrOf);
```

Comment:
What about e.g.
```c++
void f() {}
void g() { (*&f)(); }
```
Wouldn’t that run into this assertion here?

Reply:
Yes (and some tests were crashing)